recipe-client-addon/lib/EventEmitter.jsm

```diff
@@ -20,15 +19,13 @@ this.EventEmitter = function(driver) {
       Promise.resolve()
         .then(() => {
           if (!(eventName in listeners)) {
-            driver.log(`EventEmitter: Event fired with no listeners: ${eventName}`);
+            log.info(`EventEmitter: Event fired with no listeners: ${eventName}`);
             return;
           }
-          // freeze event to prevent handlers from modifying it
-          const frozenEvent = Object.freeze(event);
```

Comment:
I think in principle this looks OK from a leakiness-perspective... pushing to try to doublecheck wouldn't hurt. Just push to linux64 for only debug builds and only mochitest-bc and mochitest-e10s-bc tests and you should have an answer in relatively short order.

Separately though, you could still call `emit()` by eval'ing it inside the sandbox. It's a bit of a faff though, so if this works and offers the functionality we need, great.

Though, thinking about this more... you now have 2 emitters, and you call `emit()` on the one that lives outside the sandbox. But that one won't have the handlers of the one inside the sandbox, right? I though the idea was that code inside a recipe could listen for events from the heartbeat infra? Am I wrong about that, or wrong about how we forward the handling of events, or something else? It looks like the heartbeat tests are using the heartbeat's `eventEmitter` property, not the sandboxed one, so they will keep working, but if I'm right then 'real' recipes might be broken.